pkg/log/locallogger/dedup_handler.go

```diff
@@ -0,0 +1,280 @@
+	}
+
+	return true, entry.count // Skip this duplicate
+}
+
+// triggerCleanupIfNeeded triggers background cleanup if needed
+func (dh *DedupHandler) triggerCleanupIfNeeded(now time.Time) {
+	select {
+	case dh.cleanupTrigger <- struct{}{}:
+		// Cleanup triggered successfully
+	default:
+		// Cleanup already in progress, skip
+	}
+}
+
+// startBackgroundCleanup starts the background cleanup goroutine using gowrapper
+func (dh *DedupHandler) startBackgroundCleanup() {
+	gowrapper.Go(dh.ctx, dh.slogger, func() {
+		dh.backgroundCleanup()
+	})
+}
+
+// backgroundCleanup runs in a separate goroutine and handles cache cleanup
+func (dh *DedupHandler) backgroundCleanup() {
+	defer close(dh.cleanupDone)
+
+	for {
+		select {
+		case _, ok := <-dh.cleanupTrigger:
+			if !ok {
+				// Channel was closed, time to shutdown
+				dh.slogger.Log(dh.ctx, slog.LevelDebug,
+					"background cleanup goroutine shutting down (trigger channel closed)",
+				)
+				return
+			}
+			dh.performCleanup()
+		case <-dh.ctx.Done():
+			dh.slogger.Log(dh.ctx, slog.LevelDebug,
+				"background cleanup goroutine shutting down (context cancelled)",
+			)
+			return
+		}
+	}
+}
+
+// performCleanup removes expired entries from the deduplication cache
+func (dh *DedupHandler) performCleanup() {
+	now := time.Now()
+
+	dh.dedupMutex.Lock()
+	defer dh.dedupMutex.Unlock()
+
+	// Check if cleanup is actually needed
+	if now.Sub(dh.lastCleanup) <= cleanupInterval {
+		return
+	}
+
+	// Update cleanup timestamp
+	dh.lastCleanup = now
+
+	// Remove expired entries
+	expiredCount := 0
+	for hash, entry := range dh.dedupCache {
+		if now.Sub(entry.lastSeen) > dh.cacheExpiry {
+			delete(dh.dedupCache, hash)
+			expiredCount++
+		}
+	}
+
+	// If cache is still too large, remove oldest entries
+	if len(dh.dedupCache) > dh.maxCacheSize {
+		// Create slice of entries with their last seen times
+		type hashTime struct {
+			hash     string
+			lastSeen time.Time
+		}
+		var entries []hashTime
+		for hash, entry := range dh.dedupCache {
+			entries = append(entries, hashTime{hash, entry.lastSeen})
+		}
+
+		// Sort by last seen time (oldest first)
+		sort.Slice(entries, func(i, j int) bool {
+			return entries[i].lastSeen.Before(entries[j].lastSeen)
+		})
+
+		// Remove oldest entries to get down to maxCacheSize
+		toRemove := len(entries) - dh.maxCacheSize
+		for i := 0; i < toRemove; i++ {
+			delete(dh.dedupCache, entries[i].hash)
+		}
+
+		dh.slogger.Log(dh.ctx, slog.LevelDebug,
+			"deduplication cache cleanup completed",
+			"expired_entries", expiredCount,
+			"removed_oldest", toRemove,
+			"remaining_entries", len(dh.dedupCache),
+		)
+	} else if expiredCount > 0 {
+		dh.slogger.Log(dh.ctx, slog.LevelDebug,
+			"deduplication cache cleanup completed",
+			"expired_entries", expiredCount,
+			"remaining_entries", len(dh.dedupCache),
+		)
+	}
+}
+
+// cleanupCache is a wrapper that calls performCleanup with proper locking
+func (dh *DedupHandler) cleanupCache(now time.Time) {
```

Comment:
what's the purpose for wrapping this? vs just using triggerCleanupIfNeeded from callers?